plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDecimalCoercers.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.Decimals;
+import io.trino.spi.type.Type;
+import org.testng.annotations.Test;
+
+import java.math.BigDecimal;
+
+import static io.trino.plugin.hive.HiveTimestampPrecision.NANOSECONDS;
+import static io.trino.plugin.hive.HiveType.toHiveType;
+import static io.trino.plugin.hive.coercions.CoercionUtils.createCoercer;
+import static io.trino.spi.predicate.Utils.blockToNativeValue;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDecimalCoercers
+{
+    private static final DecimalType LONG_DECIMAL_TYPE = DecimalType.createDecimalType(30, 12);
+    private static final DecimalType SHORT_DECIMAL_TYPE = DecimalType.createDecimalType(10, 5);
+
+    @Test
+    public void testLongDecimalToTinyIntCoercion()
+    {
+        assertDecimalToIntCoercion(LONG_DECIMAL_TYPE, Decimals.valueOf(new BigDecimal("12.120000000000")), TINYINT, 12L);
+        assertDecimalToIntCoercion(LONG_DECIMAL_TYPE, Decimals.valueOf(new BigDecimal("-12.120000000000")), TINYINT, -12L);
+    }
+
+    @Test
+    public void testShortDecimalToTinyIntCoercion()
+    {
+        assertDecimalToIntCoercion(SHORT_DECIMAL_TYPE, 1212000L, TINYINT, 12L);
+        assertDecimalToIntCoercion(SHORT_DECIMAL_TYPE, -1212000L, TINYINT, -12L);
```

Comment:
nit: Can we try something like `12.12000` though we are passing an unscaled value - at times we could interpret them as a number which exceeds the range of specific type -

Reply:
Other approach we could try is to use `Decimals#parse` where we could pass a plain String and it would compute the decimal type based on the String i.e Short/Long Decimal and we could move all these test cases into a parameterized test - Have thing parameters like - `String decimalString, Type coercedType, Object expectedValue`. WDYT ?